components/locid/src/helpers.rs

```diff
@@ -69,6 +146,85 @@ impl<T> Default for ShortVec<T> {
     }
 }
 
+impl<K, V> Store<K, V> for ShortVec<(K, V)> {
+    #[inline]
+    fn lm_len(&self) -> usize {
+        self.len()
+    }
+
+    #[inline]
+    fn lm_is_empty(&self) -> bool {
+        matches!(self, ShortVec::Empty)
+    }
+
+    #[inline]
+    fn lm_get(&self, index: usize) -> Option<(&K, &V)> {
+        self.as_slice().get(index).map(|elt| (&elt.0, &elt.1))
+    }
+
+    #[inline]
+    fn lm_last(&self) -> Option<(&K, &V)> {
+        match self {
+            ShortVec::Empty => None,
+            ShortVec::Single(v) => Some(v),
+            ShortVec::Multi(v) => v.as_slice().last(),
+        }
+        .map(|elt| (&elt.0, &elt.1))
+    }
+
+    #[inline]
+    fn lm_binary_search_by<F>(&self, mut cmp: F) -> Result<usize, usize>
+    where
+        F: FnMut(&K) -> core::cmp::Ordering,
+    {
+        self.as_slice().binary_search_by(|(k, _)| cmp(k))
+    }
+}
+
+impl<K, V> StoreMut<K, V> for ShortVec<(K, V)> {
+    fn lm_with_capacity(_capacity: usize) -> Self {
+        ShortVec::Empty
+    }
+
+    // ShortVec supports reserving capacity for additional elements only if we have already allocated a vector
+    fn lm_reserve(&mut self, additional: usize) {
+        if let ShortVec::Multi(ref mut v) = self {
+            v.reserve(additional)
+        }
+    }
+
+    fn lm_get_mut(&mut self, index: usize) -> Option<(&K, &mut V)> {
+        self.as_mut_slice()
+            .get_mut(index)
+            .map(|elt| (&elt.0, &mut elt.1))
+    }
+
+    fn lm_push(&mut self, key: K, value: V) {
+        self.push((key, value))
+    }
+
+    fn lm_insert(&mut self, index: usize, key: K, value: V) {
+        self.insert(index, (key, value))
+    }
+
+    fn lm_remove(&mut self, index: usize) -> (K, V) {
+        self.remove(index)
+    }
+
+    fn lm_clear(&mut self) {
+        self.clear();
+    }
+}
+
+impl<'a, K: 'a, V: 'a> StoreIterable<'a, K, V> for ShortVec<(K, V)> {
+    type KeyValueIter =
+        core::iter::Map<core::slice::Iter<'a, (K, V)>, for<'r> fn(&'r (K, V)) -> (&'r K, &'r V)>;
+
+    fn lm_iter(&'a self) -> Self::KeyValueIter {
+        self.as_slice().iter().map(|elt| (&elt.0, &elt.1))
+    }
+}
+
```

Comment:
Nit: Add a test for your custom LiteMap store impl by using the `litemap::testing::check_litemap` function:

https://unicode-org.github.io/icu4x-docs/doc/litemap/testing/fn.check_litemap.html

Suggestion: Remove the argument of that function and instead use the new const new_empty function suggested earlier.

Reply:
^ Please use the `check_litemap` function to test your new Store impl. Also, please change the signature of that function to remove the argument and utilize the new `const EMPTY`.

Reply:
Splitting the `check_litemap` function as discussed with @robertbastian and changed the `check_litemap` signature to remove the argument.

Reply:
Added a test for `ShortVec` impl.